### PR TITLE
Add tap()

### PR DESCRIPTION
**Pull request by nohli**
```dart
    final button = spotSingle<ElevatedButton>();
    act.tap(button);
```

### CODE REVIEW

File: test/touch_event_test.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            child: null,
+          ),
+        ),
+      ),
+    );
+
+    final button = spotSingle<MaterialApp>().spotSingle<ElevatedButton>();
+
+    expect(i, 0);
+    button.tap();
```
Comment (passsy) [inline test/touch_event_test.dart:24]: I had the idea to split it into `spot` and `act`.

Then we would write

```dart
final button = spot<ElevatedButton>();
act.tap(button);
```

**Comment by passsy** [comment]
I'm quite happy with this :)

File: lib/src/act.dart [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      // covered by another widget, or outside the viewport.
+      _pokeRenderObject(
+        position: centerPosition,
+        target: renderObject,
+        snapshot: snapshot,
+      );
+
+      final binding = WidgetsBinding.instance;
+
+      // Finally, tap the widget by sending an down and up event.
```
Comment (nohli) [inline lib/src/act.dart:60]:
```suggestion
      // Finally, tap the widget by sending a down and up event.
```

File: lib/src/act.dart [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        }
+      }
+      return null;
+    }).toList();
+
+    final Element commonAncestor =
+        findCommonAncestor([elements.first, snapshot.discoveredElement!]);
+
+    throw TestFailure(
+      "Widget '${snapshot.selector.toStringWithoutParents()}' is covered by ${elements.first.widget.toStringShort()} and can't be tapped.\n"
```
Comment (nohli) [inline lib/src/act.dart:126]:
```suggestion
      "Widget '${snapshot.selector.toStringWithoutParents()}' is covered by '${elements.first.widget.toStringShort()}' and can't be tapped.\n"
```

File: test/act_test.dart [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+          ],
+        ),
+      ),
+    );
+
+    final button = spotSingle<ElevatedButton>()..existsOnce();
+    expect(
+      () => act.tap(button),
+      throwsSpotErrorContaining([
+        "Widget 'ElevatedButton' is covered by ColoredBox",
```
Comment (nohli) [inline test/act_test.dart:106]:
```suggestion
        "Widget 'ElevatedButton' is covered by 'ColoredBox'",
```

**Pull request merged**